### PR TITLE
Explicitly sort routes to reduce a chance of conflicts

### PR DESCRIPTION
**Pull request by kravets-levko**
## What type of PR is this? (check all applicable)

- [x] Other

## Description

Make sure that router will first try to match more specific routes (longer and without params), and only then will try routes with params.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,6 +64,7 @@
     "mousetrap": "^1.6.1",
     "mustache": "^2.3.0",
     "numeral": "^2.0.6",
+    "path-to-regexp": "^3.1.0",
```
Comment (kravets-levko) [inline package.json:67]: This version is used by `universal-router`

File: client/app/services/routes.js [hunk 2]
```diff
@@ -1,19 +1,40 @@
-import { isString, isObject, filter } from "lodash";
+import { isString, isObject, filter, sortBy, some } from "lodash";
+import pathToRegexp from "path-to-regexp";
+
+function isSimplePath(path) {
+  const tokens = pathToRegexp.parse(path);
+  return some(tokens, isObject);
+}
```
Comment (gabrieldutra) [inline client/app/services/routes.js:7]: This seems to be with the opposite logic

e.g.:
`/` is showing as not a simple path and `/alerts/:alertId` is showing as a simple path

Comment (kravets-levko) [reply client/app/services/routes.js]: @gabrieldutra https://github.com/getredash/redash/pull/5020/commits/e45068a0ac0b289767b3ccf3aaba3eaf75c8c58e

**Comment by kravets-levko** [comment]
@gabrieldutra I see your concerns. But let's think what kind of conflicts may be there? ATM that's `/path/:id` vs. `/path/new` - they are perfectly solved if static route will be checked first. Cases like `/path/:id` vs. `/path/:id--some-text` or `/path/:id-(.*)` could be also solved with this approach. Also, most of the time routes have a different prefix which allows to identify them. But reordering items manually is error-prone, and also keep in mind that routes may be unregistered/re-registered now, which will change manual order. So we have either to order them automatically, or introduce some value to change order of routes, like we did for settings tabs. I don't know which one is better, please let me know your opinion.

**Pull request merged**